speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp

```diff
@@ -238,6 +238,22 @@ TEST(MediatorTests, UserIDTest)
     }
 }
 
+/// @brief Tests if the Mediator sets and gets the compression level correctly
+TEST(MediatorTests, CompressionLevelTest)
+{
+    Random random;
+    char buf[32];
+    for (int i = 0; i < TEST_AMOUNT; i++)
+    {
+        SDAConfigMediator::ClearInstance();
+        ASSERT_TRUE(SetupSingletonsFolder());
+        int compressionLevel = random.NextInt();
+        SDAConfigMediator::GetInstance()->SetCompressionLevel(compressionLevel);
```

Comment:
Should it make sure that compression level throws an error if it's not uneven or it is negative?